Storage miner crash

**Describe the bug**
A miner running for ~2 days crashed unexpectedly

**Version (run `lotus version`):**
[593a7b62(next)](https://github.com/filecoin-project/lotus/tree/593a7b62) merged into `ntwk-calibration`

**Additional context**
```
2020-08-06T14:40:00.290Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:40:06.005Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:40:36.005Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:41:06.344Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:41:30.582Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:41:36.004Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:42:06.396Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:42:30.197Z        INFO    sectors storage-fsm@v0.0.0-20200730122205-d423ae90d8d4/states_sealing.go:257    scheduling seal proof computation...
2020-08-06T14:42:30.197Z        INFO    sectors storage-fsm@v0.0.0-20200730122205-d423ae90d8d4/states_sealing.go:259    KOMIT 5 017b2315e6dbf1b166f7852f36c772b50bf365cdbbe94962bf97b0b55730884f(14761); 07d059812396c0e9a3263a81b911acbb5af46a2b182d9670ad5a8955a7587abc(15815); [{131072 baga6ea4seaqandivwl2h2xpkmohd7lfyfphgci3s56v6het7esv5z4qx3cly4li} {131072 baga6ea4seaqi5hreap5iqthwen7wbxzf7a7oidokt3mht23pmnjncuee6wwq2py} {262144 baga6ea4seaqhklmwsp5bm5jehfkhnyyxvgcyb4aji6x3piyfidlclkjjdtasuby} {524288 baga6ea4seaqhaixwb57pnlp2c4ixuutbtyym5kbmnadvvxy4mz3yn3cqn3xs2gi} {1048576 baga6ea4seaqntgehxfzvoouw4ejzgzcsg3axwh2moa2noi6hvgpxbg5u3jqrmky} {2097152 baga6ea4seaqnbnjq3oylj4s4luxsukg752aiwu2bfibjghyyysm7lisubbvrgjq} {4194304 baga6ea4seaqijqccdoqgqwqbx54vui2eazh6ijf5kku5eq3xwokp6tclivuoqei} {8388608 baga6ea4seaqgl4u6lwmnerwdrm4iz7ag3mpwwaqtapc2fciabpooqmvjypweeha} {16777216 baga6ea4seaqkejdvbaufquewln7dgszre6ymaqvr2bdnyvcaee3we7gypgoocoq} {33554432 baga6ea4seaqnv7nlnwutmrctyjwtg4tlt7x6gq56r6awjhwabgvnh6x7kbqxkca} {67108864 baga6ea4seaqnsqov4dldcsuzlqz77pkpxzurddlt2ts72lgtd4hxzbxl3ukoobq} {134217728 baga6ea4seaqfctcdlq6qju2juu3f7pkz77drgyurcf4fteobupctv4rapf2buly} {268435456 baga6ea4seaqk2bufhfu5g7ju74eobh2wsmfevum2rhppmdf75z7b2m4byhtryny}]; r:626167626f6561346235616263626470657368676933346d6b757674787368646b717279696e69366764726c6262353732687a6937706e6b75617a6a6b34736b35; d:6261676136656134736561716d7962686b6334696d703278336770726a6a376b6e6d68727a6b756f7267356e767471767178637437656c3461696b756d736971
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:286  SCHED 1 queued; 2 open windows
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:351  SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:352  SCHED Acceptable win: [[0 1]]
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:366  SCHED try assign sqi:0 sector 5 to window 0
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:373  SCHED ASSIGNED sqi:0 sector 5 to window 0
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:516  assign worker sector 5
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:286  SCHED 0 queued; 2 open windows
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:351  SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:352  SCHED Acceptable win: []
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/localworker.go:75     acquired sector {31195 5} (e:6; a:0): {{0 0}  /srv/lotus/.lotusminer/sealed/s-t031195-5 /srv/lotus/.lotusminer/cache/s-t031195-5}
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/localworker.go:75     acquired sector {31195 5} (e:6; a:0): {{0 0}  /srv/lotus/.lotusminer/sealed/s-t031195-5 /srv/lotus/.lotusminer/cache/s-t031195-5}
2020-08-06T14:42:30.197Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:489  task done       {"workerid": 0}
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:286  SCHED 1 queued; 2 open windows
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:351  SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:352  SCHED Acceptable win: [[0 1]]
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:366  SCHED try assign sqi:0 sector 5 to window 0
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:373  SCHED ASSIGNED sqi:0 sector 5 to window 0
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:516  assign worker sector 5
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:286  SCHED 0 queued; 2 open windows
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:351  SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-08-06T14:42:30.220Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:352  SCHED Acceptable win: []
2020-08-06T14:42:30.221Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:489  task done       {"workerid": 0}
2020-08-06T14:42:30.375Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:42:30.482Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 3}
2020-08-06T14:42:36.006Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:43:06.004Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:43:30.490Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:43:36.240Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:44:00.469Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:44:06.108Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:44:30.195Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:44:30.201Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 3}
2020-08-06T14:44:30.314Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 4}
2020-08-06T14:44:30.617Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 4}
2020-08-06T14:44:30.799Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 5}
2020-08-06T14:44:36.006Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:45:00.621Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:45:06.005Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:45:36.004Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 36s (nulls: 1)
2020-08-06T14:46:00.197Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:46:00.486Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 3}
2020-08-06T14:46:06.325Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:46:30.282Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:286  SCHED 1 queued; 2 open windows
2020-08-06T14:46:30.282Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:351  SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-08-06T14:46:30.282Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:352  SCHED Acceptable win: [[0 1]]
2020-08-06T14:46:30.282Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:366  SCHED try assign sqi:0 sector 5 to window 0
2020-08-06T14:46:30.282Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:373  SCHED ASSIGNED sqi:0 sector 5 to window 0
2020-08-06T14:46:30.282Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:516  assign worker sector 5
2020-08-06T14:46:30.282Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:286  SCHED 0 queued; 2 open windows
2020-08-06T14:46:30.283Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:351  SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-08-06T14:46:30.283Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:352  SCHED Acceptable win: []
2020-08-06T14:46:30.283Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/localworker.go:75     acquired sector {31195 5} (e:7; a:0): {{0 0} /srv/lotus/.lotusminer/unsealed/s-t031195-5 /srv/lotus/.lotusminer/sealed/s-t031195-5 /srv/lotus/.lotusminer/cache/s-t031195-5}
2020-08-06T14:46:30.283Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/localworker.go:75     acquired sector {31195 5} (e:4; a:0): {{0 0}   /srv/lotus/.lotusminer/cache/s-t031195-5}
2020-08-06T14:46:30.283Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:489  task done       {"workerid": 0}
2020-08-06T14:46:30.310Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:46:30.507Z        INFO    stores  stores/local.go:500     remove /srv/lotus/.lotusminer/unsealed/s-t031195-5
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:286  SCHED 1 queued; 2 open windows
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:351  SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:352  SCHED Acceptable win: [[0 1]]
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:366  SCHED try assign sqi:0 sector 5 to window 0
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:373  SCHED ASSIGNED sqi:0 sector 5 to window 0
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:516  assign worker sector 5
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:286  SCHED 0 queued; 2 open windows
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:351  SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:352  SCHED Acceptable win: []
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/localworker.go:75     acquired sector {31195 5} (e:6; a:0): {{0 0}  /srv/lotus/.lotusminer/sealed/s-t031195-5 /srv/lotus/.lotusminer/cache/s-t031195-5}
2020-08-06T14:46:30.553Z        DEBUG   stores  stores/local.go:536     not moving {31195 5}(2); src and dest are the same
2020-08-06T14:46:30.553Z        DEBUG   stores  stores/local.go:536     not moving {31195 5}(4); src and dest are the same
2020-08-06T14:46:30.553Z        DEBUG   advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/sched.go:489  task done       {"workerid": 0}
2020-08-06T14:46:30.557Z        INFO    sectors storage-fsm@v0.0.0-20200730122205-d423ae90d8d4/states_sealing.go:368    Proving sector 5
2020-08-06T14:46:30.557Z        WARN    advmgr  sector-storage@v0.0.0-20200803184904-3cab915fd225/manager.go:469        ReleaseUnsealed todo
2020-08-06T14:46:36.029Z        INFO    miner   miner/miner.go:307      Time delta between now and our mining base: 6s (nulls: 0)
2020-08-06T14:47:00.302Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 2}
2020-08-06T14:47:00.522Z        INFO    rpc     go-jsonrpc@v0.1.1-0.20200602181149-522144ab4e24/client.go:204   rpc output message buffer       {"n": 3}
2020-08-06T14:47:00.530Z        INFO    storageadapter  storageadapter/provider.go:283  Storage deal 9527 activated at epoch 15823
2020-08-06T14:47:00.530Z        INFO    markets loggers/loggers.go:18   storage event   {"name": "ProviderEventDealActivated", "proposal CID": "bafyreifirqbky6fnoha4yieizpjlbnmijvz577uwuej3xz5z5jc2hvnjma", "state": "StorageDealFinalizing", "message": ""}
2020-08-06T14:47:00.532Z        INFO    markets loggers/loggers.go:18   storage event   {"name": "ProviderEventCleanupFinished", "proposal CID": "bafyreifirqbky6fnoha4yieizpjlbnmijvz577uwuej3xz5z5jc2hvnjma", "state": "StorageDealActive", "message": ""}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x153bf62]

goroutine 9001042 [running]:   
github.com/filecoin-project/lotus/chain/types.(*TipSet).Height(...)
        /srv/lotus/lotus/chain/types/tipset.go:145
github.com/filecoin-project/lotus/markets/storageadapter.(*ProviderNodeAdapter).OnDealExpiredOrSlashed.func1(0x0, 0x10, 0x2701280, 0x520066)
        /srv/lotus/lotus/markets/storageadapter/provider.go:387 +0x32
github.com/filecoin-project/lotus/chain/events.(*hcEvents).onHeadChanged(0xc00055e480, 0xc01496af40, 0xc0124ce460, 0x2ae00c0, 0x6, 0xac45a, 0x0, 0x0, 0x0)
        /srv/lotus/lotus/chain/events/events_called.go:311 +0xc9
github.com/filecoin-project/lotus/chain/events.(*watcherEvents).StateChanged(0xc00055e538, 0xc01496af40, 0xc000126b70, 0x2ae00c0, 0x6, 0xac45a, 0xc01496b040, 0x0, 0x0)
        /srv/lotus/lotus/chain/events/events_called.go:440 +0xc4
github.com/filecoin-project/lotus/markets/storageadapter.(*ProviderNodeAdapter).OnDealExpiredOrSlashed(0xc000220a50, 0x2d44680, 0xc00003e0f0, 0x2537, 0xc014473e80, 0xc014473ea0, 0x19fad10e001a4240, 0x54948e0)
        /srv/lotus/lotus/markets/storageadapter/provider.go:456 +0x5ae
github.com/filecoin-project/go-fil-markets/storagemarket/impl/providerstates.WaitForDealCompletion(0x7fe24c036748, 0xc011bdb710, 0x7fe24c036770, 0xc005ab83f8, 0xc01ca46a20, 0x27, 0x20000, 0x0, 0xc017d5f920, 0x15, ...)
        /srv/lotus/go/pkg/mod/github.com/filecoin-project/go-fil-markets@v0.5.3/storagemarket/impl/providerstates/provider_states.go:408 +0x120
reflect.Value.call(0x26660c0, 0x2adfa28, 0x13, 0x292e1b6, 0x4, 0xc010dc9710, 0x3, 0x3, 0x51daaf, 0x27e7620, ...)
        /usr/local/go/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0x26660c0, 0x2adfa28, 0x13, 0xc015f56710, 0x3, 0x3, 0x2d72b60, 0x2, 0x4b)
        /usr/local/go/src/reflect/value.go:321 +0xb4
github.com/filecoin-project/go-statemachine/fsm.fsmHandler.handler.func1(0xc000126150, 0x2, 0x2, 0xc000126150, 0xc015f56b48, 0x756ea18235764c)
        /srv/lotus/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200730031800-c3336614d2a7/fsm/fsm.go:170 +0x424
reflect.callReflect(0xc0001260f0, 0xc010ae19e0, 0xc010ae19c8)
        /usr/local/go/src/reflect/value.go:549 +0x322
reflect.makeFuncStub(0x2d44680, 0xc00003e0f0, 0xc0190cbe40, 0xc01ca46a20, 0x27, 0x20000, 0x0, 0xc017d5f920, 0x15, 0xc013679f78, ...)
        /usr/local/go/src/reflect/asm_amd64.s:20 +0x42
reflect.Value.call(0xc013fa8720, 0xc0001260f0, 0x13, 0x292e1b6, 0x4, 0xc015f56f70, 0x2, 0x2, 0x51daaf, 0x272d7c0, ...)
        /usr/local/go/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0xc013fa8720, 0xc0001260f0, 0x13, 0xc01395f770, 0x2, 0x2, 0x0, 0x8000106, 0x0)
        /usr/local/go/src/reflect/value.go:321 +0xb4
github.com/filecoin-project/go-statemachine.(*StateMachine).run.func3(0xc0190cbdb0, 0x2d44680, 0xc00003e0f0, 0xc0190cbe40, 0xc0190cbdc0, 0xc014e5b500)
        /srv/lotus/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200730031800-c3336614d2a7/machine.go:102 +0x2bf
created by github.com/filecoin-project/go-statemachine.(*StateMachine).run
        /srv/lotus/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200730031800-c3336614d2a7/machine.go:100 +0x3da
```

--- a/markets/storageadapter/provider.go
+++ b/markets/storageadapter/provider.go
@@ -379,6 +379,11 @@ func (n *ProviderNodeAdapter) OnDealExpiredOrSlashed(ctx context.Context, dealID
 
 	// Called immediately to check if the deal has already expired or been slashed
 	checkFunc := func(ts *types.TipSet) (done bool, more bool, err error) {
+		if ts == nil {
+			// keep listening for events
+			return false, true, nil
+		}
+
 		// Check if the deal has already expired
 		if sd.Proposal.EndEpoch <= ts.Height() {
 			onDealExpired(nil)

